cloudferry_devlab/cloudferry_devlab/tests/testcases/test_keystone_migration.py

```diff
@@ -34,24 +36,43 @@ def test_migrate_keystone_users(self, param):
         :param name: user's name
         :param description: user's description
         :param enabled: is user enabled"""
-        self.validate_resource_parameter_in_dst(self.src_users, self.dst_users,
+        filtered_src_users = [user for user in self.src_users
+                              if getattr(user, "name") !=
+                              config.case_sensitivity_test_user]
+        self.validate_resource_parameter_in_dst(filtered_src_users,
+                                                self.dst_users,
                                                 resource_name='user',
                                                 parameter=param)
 
-    @attr(migrated_tenant=['admin', 'tenant1', 'tenant2'])
+    @attr(migrated_tenant=['admin', 'tenant1', 'tenant2', 'tenant6'])
     def test_migrate_keystone_user_tenant_roles(self):
         """Validate user's tenant roles were migrated with correct name."""
-        src_user_names = [user.name for user in self.filter_users()]
+        tenant = base.get_nosetest_cmd_attribute_val('migrated_tenant')
         for dst_user in self.dst_users:
-            if dst_user.name not in src_user_names:
+            src_user = None
+            for user in self.src_users:
+                if user.name.lower() == dst_user.name.lower():
+                    src_user = user
+                    break
+            if src_user is None:
+                continue
+            if tenant is None:
+                src_user_tnt_roles = self.src_cloud.get_user_tenant_roles(
+                                         src_user)
+                dst_user_tnt_roles = self.dst_cloud.get_user_tenant_roles(
+                                         dst_user)
+            else:
+                src_user_tnt_roles = self.src_cloud.get_roles_for_user(
+                                         src_user, tenant)
+                dst_user_tnt_roles = self.dst_cloud.get_roles_for_user(
+                                         dst_user, tenant)
+            if len(src_user_tnt_roles) == 0 and len(dst_user_tnt_roles) == 0:
```

Comment:
Actually I received this comment earlier also from you. This is necessary because the validate function will skip the further user tenant role verification (the way they are implemented right now) if any of user tenant roles do not exist on the source as well as destination clouds while we are iterating on the destination users list. 
And the tenant conditional check is to enable this test to be run for verifying the user tenant roles for particular tenant and then for all tenants. And as per your another comment earlier, I wrote down a separate function for retrieving user tenant roles for a particular tenant.